packages/token/src/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.15.0 (token/erc2981/erc2981.cairo)
+
+/// #ERC2981 Component
+///
+/// Implementation of the NFT Royalty Standard, a standardized way to retrieve royalty payment
+/// information.
+///
+/// Royalty information can be specified globally for all token ids via `set_default_royalty`,
+/// and/or individually for specific token ids via `set_token_royalty`. The latter takes precedence
+/// over the first.
+///
+/// Royalty is specified as a fraction of sale price. The denominator is set by the contract by
+/// using the Immutable pattern.
+///
+/// IMPORTANT: ERC-2981 only specifies a way to signal royalty information and does not enforce its
+/// payment. See https://eips.ethereum.org/EIPS/eip-2981#optional-royalty-payments[Rationale] in the
+/// ERC. Marketplaces are expected to voluntarily pay royalties together with sales.
+#[starknet::component]
+pub mod ERC2981Component {
+    use core::num::traits::Zero;
+    use openzeppelin_introspection::src5::SRC5Component::InternalTrait as SRC5InternalTrait;
+    use openzeppelin_introspection::src5::SRC5Component::SRC5Impl;
+    use openzeppelin_introspection::src5::SRC5Component;
+    use openzeppelin_token::common::erc2981::interface::{IERC2981, IERC2981_ID};
+    use starknet::ContractAddress;
+    use starknet::storage::Map;
+
+    // This default denominator is only used when the
+    pub const DEFAULT_FEE_DENOMINATOR: u256 = 10_000;
+
+    #[derive(Serde, Drop, starknet::Store)]
+    struct RoyaltyInfo {
+        pub receiver: ContractAddress,
+        pub royalty_fraction: u256,
```

Comment:
In Solidity u96 is used to fit both struct members into a single storage slot. We can use u128 since u256 uses two storage slots so is also saving storage space. I will update.